validation/scripts/test_discovery.py

```diff
@@ -39,33 +42,65 @@ def get_test_data():
 
     for test in tests:
         # layers/data tuples
-        layers, data = get_data(test)
+        layers_file, data_file = get_data(test)
 
-        slabs = np.loadtxt(layers)
+        slabs = np.loadtxt(os.path.join(PTH, layers_file))
         assert slabs.shape[1] == 4
 
-        data = np.loadtxt(data)
+        data = np.loadtxt(os.path.join(PTH, data_file))
         assert data.shape[1] in [2, 3, 4]
 
         yield os.path.basename(test), slabs, data
 
 
 def get_data(file):
-    # for each of the unpolarised test files figure out where the data and
-    # layer parameters are
-
-    with open(file, "r") as f:
-        # ignore comment lines starting with # or space
-        while True:
-            line = f.readline()
-            if line.lstrip(" \t").startswith("#"):
-                continue
-            elif line == "\n":
-                continue
-            else:
-                layers = line.rstrip("\n")
-                data = f.readline().rstrip("\n")
-
-                layers = os.path.join(PTH, layers)
-                data = os.path.join(PTH, data)
-                return (layers, data)
+    # for each of the test files extract the parameters,
+    # e.g. the names of the data_file and layers_file
+
+    with open(file, "rt") as f:
+        # ignore comment lines starting with #
+        for line in f:
+            line = line.strip()
+            if not line.startswith("#"):
+                yield line
+
+
+def get_polarised_test_data():
+    """
+    A generator yielding (slabs, data) tuples.
```

Comment:
The yield is `(slabs, data, AGUIDE, H)` tuples.

Reply:
See [9f9101c](https://github.com/reflectivity/analysis/pull/66/commits/9f9101c753d8b4b2e9ff4386e6a9b9873d82d147)